typed-racket-lib/typed-racket/base-env/prims-struct.rkt

```diff
@@ -81,12 +81,22 @@
                                               (~bind [ecname #f]))
                                         (~and (~seq #:extra-constructor-name ecname)
                                               (~bind [cname #f]))))
+                        (~optional (~seq #:type-name type:id))
                         ;; FIXME: unsound, but relied on in core libraries
                         ;; #:guard ought to be supportable with some work
                         ;; #:property is harder
                         (~optional (~seq #:guard guard:expr))
                         (~seq #:property prop:expr prop-val:expr))
-                   ...)))
+                   ...)
+             #:attr untyped #`(#,@(if (attribute mutable?) #'(#:mutable) #'())
```

Comment:
@samth This bit right here is sort of icky. I’d appreciate any tips on how to do this without the brittle repetition.